contrib/confluentinc/confluent-kafka-go/kafka/kafka_test.go

```diff
@@ -277,3 +277,49 @@ func TestDeprecatedContext(t *testing.T) {
 	<-c.Events()
 
 }
+
+func TestCustomTags(t *testing.T) {
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	c, err := NewConsumer(&kafka.ConfigMap{
+		"go.events.channel.enable": true, // required for the events channel to be turned on
+		"group.id":                 testGroupID,
+		"socket.timeout.ms":        10,
+		"session.timeout.ms":       10,
+		"enable.auto.offset.store": false,
+	}, WithCustomTag("foo", func(msg *kafka.Message) interface{} {
+		return "bar"
+	}), WithCustomTag("key", func(msg *kafka.Message) interface{} {
+		return msg.Key
+	}))
+	assert.NoError(t, err)
+
+	err = c.Subscribe(testTopic, nil)
+	assert.NoError(t, err)
+
+	go func() {
+		c.Consumer.Events() <- &kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &testTopic,
+				Partition: 1,
+				Offset:    1,
+			},
+			Key:   []byte("key1"),
+			Value: []byte("value1"),
+		}
+	}()
+
+	_ = (<-c.Events()).(*kafka.Message)
```

Comment:
```suggestion
	<-c.Events()
```
Why does it need a cast and a `_`? Or are you trying to test the type of the returned value, in which case ignore my comment.

Reply:
the object from the channel is not consumed and what you suggested is preferred.